copyfromstaging.sh

```diff
@@ -0,0 +1,104 @@
+  fi
+
+  # All source files need the extension .g.cs.  Some will only be generated with the
+  # extension .cs.  Rename files that end only in .cs.
+  for cs_file in $(find "$STAGING_DIR" -name '*.cs'); do
+    if [[ $cs_file != *.g.cs ]]; then
+      mv $cs_file $(echo $cs_file | sed -E 's/.cs$/.g.cs/g')
+    fi
+  done
+
+  # Remove the newly generated standalone snippets until they are ready for surfacing.
+  rm -rf "$STAGING_DIR/$1.StandaloneSnippets"
+
+  # We generate our own project files
+  rm -f $(find "$STAGING_DIR" -name '*.csproj')
+
+  # Wipe out all the existing source files we're about to replace.
+  # This will clear out source files that are no longer generated.
+  # BAD because this deleted GrpcCtorCompatibility.g.cs.
+  # rm -f $(find "$PACKAGE_DIR" -name '*.cs')
+
+  # Copy the files out of the staging directory into the package directory.
+  mkdir -p "$PACKAGE_DIR"
+  cp -rf $STAGING_DIR/* "$PACKAGE_DIR"
+
+  # Clean out the staging directory.
+  rm -rf "$STAGING_DIR"
+
+  # The following code was copied from generate_api().
+  if [[ -f $PACKAGE_DIR/postgeneration.patch ]]; then
+    echo "Applying post-generation patch for $PACKAGE"
+    (
+      cd $PACKAGE_DIR
+      git apply postgeneration.patch
+    )
+  fi
+
+  if [[ $(grep -E "^namespace" apis/$1/$1/*.cs | grep -Ev "namespace ${1}[[:space:]{]*\$") ]]; then
+    # We know Google.LongRunning contains a proto in Google.Cloud.
+    if [[ $1 == "Google.LongRunning" ]]; then
+      echo "Ignoring broken namespaces in $1"
+    else
+      echo "API $1 has broken namespace declarations"
+      exit 1
+    fi
+  fi
+
+  if [[ $CHECK_COMPATIBILITY == "true" ]]
+  then
+    if [[ -f $TMP/$PACKAGE.dll ]]
+    then
+      # In order to skip expensive "build and check compatibility" step,
+      # first see whether git thinks anything has changed.
+      # Command line arguments:
+      # -c core.safecrlf=false: don't warn on line ending changes
+      # diff: do a diff :)
+      # -s: suppress output as we only care about the exit code
+      # -b: ignore whitespace
+      # --exit-code: set the exit code to 0 for no diff, 1 when a diff is detected      
+      if git -c core.safecrlf=false diff -s -b --exit-code -- $PACKAGE_DIR
+      then
+        echo "git detects no change in $PACKAGE; skipping compatibility checking"
+      else
+        echo "Building new version of $PACKAGE for compatibility checking"
+        dotnet build -c Release -f netstandard2.0 -v quiet -nologo -clp:NoSummary -p:SourceLinkCreate=false $PACKAGE_DIR/$PACKAGE
+        echo ""
+        echo "Changes in $PACKAGE:"
+        dotnet run -p tools/Google.Cloud.Tools.CompareVersions -- \
+          --file1=$TMP/$PACKAGE.dll \
+          --file2=$PACKAGE_DIR/$PACKAGE/bin/Release/netstandard2.0/$PACKAGE.dll
+      fi
+    else
+      echo ""
+      echo "$PACKAGE is a new API."
+    fi
+  fi
+}
+
+# Iterate over all the apis in the owl-bot-staging directory.
+if [[ -d owl-bot-staging ]]
```

Comment:
So for a mono repo, OwlBot uses separate OwlBot.yaml files, but then a single post-process script? Or will this actually only ever get a single API at a time, and it's just a matter of finding it?

(If it's all of them, does that mean we'd have multiple APIs changed in a single commit?)

Reply:
The plan is specify the post-processor docker image in the usual `.github/.OwlBot.yaml` location.  That one post processor will run across the whole repo.

Reply:
Sorry, I wasn't clear - it's about how often it's run rather than the location. Basically, I'm hoping we only end up dealing with a single API per commit. If we end up with two very closely related APIs (e.g. OSConfig.V1 and OSConfig.V1Alpha) in the same commit, that's okay.

Reply:
The post processor will run on every pull request, while the pull request is still open.

Reply:
And there's a 1:1 correspondence between a commit (or PR?) in googleapis-gen (or googleapis?) and OwlBot? That's what I'm trying to get at, eventually: will I end up with PRs containing commits to multiple APIs? I suspect not, but I'm just trying to check.